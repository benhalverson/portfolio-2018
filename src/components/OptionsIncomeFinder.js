import React, { Fragment } from 'react';
import OptionsIncomeFinderDefaultView from '../images/OptionsIncomeFinderDefaultView.png';
import OptionsIncomeFinderAdditionalFilters from '../images/OptionsIncomeFinderAdditionalFilters.png';
import OptionsIncomeFinderExpandedResults from '../images/OptionsIncomeFinderExpandedResults.png';
import OptionsIncomeFinderMiniBacktester from '../images/OptionsIncomeFinderMinitBacktester.png';
import OptionsIncomeFinderResults from '../images/OptionsIncomeFinderResults.png';
import OptionsIncomeFinderSearch from '../images/OptionsIncomeFinderSearch.png';

const OptionsIncomeFinder = () => (
  <Fragment>
    <div className="title">E-Trade Options Income Finder</div>
    <div className="body">
      <p className="description">
        I was the lead on this new product. I built the first prototype in
        Angular.js and version 1 of this product.
      </p>
      <div className="screenshots">
        <img className="col" src={OptionsIncomeFinderDefaultView} alt="" />
        <img
          className="col"
          src={OptionsIncomeFinderAdditionalFilters}
          alt=""
        />
        <img className="col" src={OptionsIncomeFinderSearch} alt="" />
        <img className="col" src={OptionsIncomeFinderResults} alt="" />
        <img className="col" src={OptionsIncomeFinderExpandedResults} alt="" />
        <img className="col" src={OptionsIncomeFinderMiniBacktester} alt="" />
      </div>
    </div>
  </Fragment>
);
export default OptionsIncomeFinder;
