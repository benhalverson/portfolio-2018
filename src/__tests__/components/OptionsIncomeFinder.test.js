import React from 'react';
import { shallow } from 'enzyme';
import OptionsIncomeFinder from '../../components/OptionsIncomeFinder';
test('it should render the OptionsIncomeFinder component', () => {
  const wrapper = shallow(<OptionsIncomeFinder/>);
  expect(wrapper).toMatchSnapshot();
});