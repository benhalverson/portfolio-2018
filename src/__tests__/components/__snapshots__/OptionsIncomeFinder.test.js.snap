// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the OptionsIncomeFinder component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OptionsIncomeFinder />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="title"
>
          E-Trade Options Income Finder
</div>,
        <div
          className="body"
>
          <p
                    className="description"
          >
                    I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.
          </p>
          <div
                    className="screenshots"
          >
                    <img
                              alt=""
                              className="col"
                              src="OptionsIncomeFinderDefaultView.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="OptionsIncomeFinderAdditionalFilters.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="OptionsIncomeFinderSearch.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="OptionsIncomeFinderResults.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="OptionsIncomeFinderExpandedResults.png"
                    />
                    <img
                              alt=""
                              className="col"
                              src="OptionsIncomeFinderMinitBacktester.png"
                    />
          </div>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "E-Trade Options Income Finder",
          "className": "title",
        },
        "ref": null,
        "rendered": "E-Trade Options Income Finder",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="description"
>
              I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.
</p>,
            <div
              className="screenshots"
>
              <img
                            alt=""
                            className="col"
                            src="OptionsIncomeFinderDefaultView.png"
              />
              <img
                            alt=""
                            className="col"
                            src="OptionsIncomeFinderAdditionalFilters.png"
              />
              <img
                            alt=""
                            className="col"
                            src="OptionsIncomeFinderSearch.png"
              />
              <img
                            alt=""
                            className="col"
                            src="OptionsIncomeFinderResults.png"
              />
              <img
                            alt=""
                            className="col"
                            src="OptionsIncomeFinderExpandedResults.png"
              />
              <img
                            alt=""
                            className="col"
                            src="OptionsIncomeFinderMinitBacktester.png"
              />
</div>,
          ],
          "className": "body",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.",
              "className": "description",
            },
            "ref": null,
            "rendered": "I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt=""
                  className="col"
                  src="OptionsIncomeFinderDefaultView.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="OptionsIncomeFinderAdditionalFilters.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="OptionsIncomeFinderSearch.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="OptionsIncomeFinderResults.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="OptionsIncomeFinderExpandedResults.png"
/>,
                <img
                  alt=""
                  className="col"
                  src="OptionsIncomeFinderMinitBacktester.png"
/>,
              ],
              "className": "screenshots",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OptionsIncomeFinderDefaultView.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OptionsIncomeFinderAdditionalFilters.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OptionsIncomeFinderSearch.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OptionsIncomeFinderResults.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OptionsIncomeFinderExpandedResults.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "",
                  "className": "col",
                  "src": "OptionsIncomeFinderMinitBacktester.png",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="title"
>
            E-Trade Options Income Finder
</div>,
          <div
            className="body"
>
            <p
                        className="description"
            >
                        I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.
            </p>
            <div
                        className="screenshots"
            >
                        <img
                                    alt=""
                                    className="col"
                                    src="OptionsIncomeFinderDefaultView.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="OptionsIncomeFinderAdditionalFilters.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="OptionsIncomeFinderSearch.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="OptionsIncomeFinderResults.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="OptionsIncomeFinderExpandedResults.png"
                        />
                        <img
                                    alt=""
                                    className="col"
                                    src="OptionsIncomeFinderMinitBacktester.png"
                        />
            </div>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "E-Trade Options Income Finder",
            "className": "title",
          },
          "ref": null,
          "rendered": "E-Trade Options Income Finder",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="description"
>
                I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.
</p>,
              <div
                className="screenshots"
>
                <img
                                alt=""
                                className="col"
                                src="OptionsIncomeFinderDefaultView.png"
                />
                <img
                                alt=""
                                className="col"
                                src="OptionsIncomeFinderAdditionalFilters.png"
                />
                <img
                                alt=""
                                className="col"
                                src="OptionsIncomeFinderSearch.png"
                />
                <img
                                alt=""
                                className="col"
                                src="OptionsIncomeFinderResults.png"
                />
                <img
                                alt=""
                                className="col"
                                src="OptionsIncomeFinderExpandedResults.png"
                />
                <img
                                alt=""
                                className="col"
                                src="OptionsIncomeFinderMinitBacktester.png"
                />
</div>,
            ],
            "className": "body",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.",
                "className": "description",
              },
              "ref": null,
              "rendered": "I was the lead on this new product. I built the first prototype in Angular.js and version 1 of this product.",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <img
                    alt=""
                    className="col"
                    src="OptionsIncomeFinderDefaultView.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="OptionsIncomeFinderAdditionalFilters.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="OptionsIncomeFinderSearch.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="OptionsIncomeFinderResults.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="OptionsIncomeFinderExpandedResults.png"
/>,
                  <img
                    alt=""
                    className="col"
                    src="OptionsIncomeFinderMinitBacktester.png"
/>,
                ],
                "className": "screenshots",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OptionsIncomeFinderDefaultView.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OptionsIncomeFinderAdditionalFilters.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OptionsIncomeFinderSearch.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OptionsIncomeFinderResults.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OptionsIncomeFinderExpandedResults.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "alt": "",
                    "className": "col",
                    "src": "OptionsIncomeFinderMinitBacktester.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "img",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
